checker.cpp

```diff
@@ -1,17 +1,130 @@
 #include <assert.h>
+#include <iostream>
+#include <map>
+#include<string>
+#include<vector>
+using namespace std;
+class Alert {
+public:
+    virtual void raiseAnAlert( string message) = 0;
 
-bool vitalsAreOk(float bpm, float spo2, float respRate) {
-  if(bpm < 70 || bpm > 150) {
-    return false;
-  } else if(spo2 < 90) {
-    return false;
-  } else if(respRate < 30 || respRate > 95) {
-    return false;
-  }
-  return true;
-}
+};
+
+class AlertInSMS : public Alert {
+public:
+    virtual void raiseAnAlert(string message) override {
+        cout << "send SMS :" << message << endl;
+    }
+};
+
+class AlertInSound : public Alert {
+public:
+    virtual void raiseAnAlert(string message) override {
+        cout << "Make Alert sound :" << message << endl;
+    }
+};
+
+class Range {
+public:
+    static bool isInRange(float value, float low, float high) {
+        return value >= low && value <= high;
+    }
+    static bool isLessThan(float value, float low) {
+        return value < low;
+    }
+    static bool isGreaterThan(float value, float high) {
+        return value > high;
+    }
+};
+
+
+struct Vital {
+    string vital_name;
+    float value;
+};
+
+class Vitals {
+public:
+    static map<string, pair<int, int>> vitals_information;
```

Comment:
Nice use of map 👍 
What do you think about mapping the string to a struct, so that the lower and upper limits can be named, instead of referring them as first and second?